packages/payload/src/admin/components/forms/field-types/Relationship/index.tsx

```diff
@@ -178,7 +183,7 @@ const Relationship: React.FC<Props> = (props) => {
             }
 
             if (filterOptionsResult?.[relation]) {
-              query.where.and.push(filterOptionsResult[relation])
+              query.where.and.push(filterOptionsResult[relation] as Where)
```

Comment:
filterOptionsResult[relation] can now also return `true`. What if it's true here? It's no `Where` in that case.

Maybe it should be

`if (filterOptionsResult?.[relation] && typeof filterOptionsResult?.[relation] !== 'boolean) {`